commands/verify/verify.go

```diff
@@ -84,165 +108,252 @@ func verifySingleCase(expectedFile, actualFile, query string) error {
 		}
 		return fmt.Errorf("failed to verify the output: %s, error:\n%v", sourceName, err)
 	}
-	logger.Log.Infof("verified the output: %s", sourceName)
 	return nil
 }
 
-// concurrentErrors store errors that occurred when verifying cases in goroutines.
-type concurrentErrors struct {
-	errs  *multierror.Error
-	mutex sync.Mutex
-}
-
-// verifyInfo contains necessary information about verification
-type verifyInfo struct {
-	retryCount int
-	interval   time.Duration
-	failFast   bool
-}
+func concurrentlyVerifySingleCase(v *config.VerifyCase, verifyInfo *verifyInfo, wg *sync.WaitGroup, outputInfo *OutputInfo, stopChan chan bool) {
+	var msg string
+	var err error
+	var caseInfo CaseInfo
+	defer func() {
+		outputInfo.writeLock.Lock()
+		outputInfo.casesInfo = append(outputInfo.casesInfo, caseInfo)
+		if caseInfo.err != nil {
+			*verifyInfo.errorExist = true
+			if verifyInfo.failFast {
+				stopChan <- true
+				return
```

Comment:
If this line is reached, then `outputInfo.writeLock.Unlock()` is never called?

Reply:
By design, to prevent other goroutines to append their `caseInfo`. A bit anti-pattern, though. 😂